pipeline/pipelines/gapic_generation_pipeline.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Pipelines that run GAPIC"""
+
+from pipeline.pipelines import pipeline_base
+from pipeline.pipelines import package_generation_pipeline
+from pipeline.tasks import gapic_tasks, format_tasks, protoc_tasks
+from taskflow.patterns import linear_flow
+
+
+class GapicConfigPipeline(pipeline_base.PipelineBase):
+
+    def __init__(self, **kwargs):
+        kwargs['language'] = ''
+        super(GapicConfigPipeline, self).__init__(**kwargs)
+
+    def do_build_flow(self, **kwargs):
+        flow = linear_flow.Flow('gapic-configgen')
+        flow.add(protoc_tasks.ProtoDescGenTask('ProtoDesc', inject=kwargs),
+                 gapic_tasks.GapicConfigGenTask(
+                     'GapicConfigGen', inject=kwargs),
+                 gapic_tasks.GapicConfigMoveTask(
+                     'GapicConfigMove', inject=kwargs))
+        return flow
+
+    def validate_kwargs(self, **kwargs):
+        pass
+
+
+class GapicClientPipeline(pipeline_base.PipelineBase):
+
+    def __init__(self, **kwargs):
+        super(GapicClientPipeline, self).__init__(**kwargs)
+
+    def do_build_flow(self, **kwargs):
+        flow = linear_flow.Flow('gapic-codegen')
+        flow.add(protoc_tasks.ProtoDescGenTask('ProtoDesc', inject=kwargs),
+                 gapic_tasks.GapicCodeGenTask('GapicCodegen', inject=kwargs),
+                 format_tasks.make_format_task(
+                     kwargs['language'], 'GapicFormat', kwargs))
+        return flow
+
+    def validate_kwargs(self, **kwargs):
+        pass
+
+
+class GapicClientPackagePipeline(pipeline_base.PipelineBase):
```

Comment:
Or I'm wondering if there's a better way to name this -- right now, I find it somewhat unintuitive that `PythonGapicClientPipeline` is a `GapicClientPackagePipeline` that has (via its flow) a `GapicClientPipeline` -- maybe there's someway to reflect this relationship better in the names.

But if not, I'm fine leaving it as is, since one of the key points of pipelines is that they may contain other pipelines, which might just naturally lead to situations like this.